gateway/src/node/client_handling/websocket/connection_handler.rs

```diff
@@ -378,12 +379,29 @@ where
                     enc_address,
                     iv,
                 } => {
+                    *final_receiver = Some(mix_receiver);
                     self.handle_authenticate(address, enc_address, iv, mix_sender)
                         .await
                 }
                 ClientControlRequest::RegisterHandshakeInitRequest { data } => {
+                    *final_receiver = Some(mix_receiver);
                     self.handle_register(data, mix_sender).await
                 }
+                ClientControlRequest::BandwidthCredential { enc_credential } => {
+                    let credential = match ClientControlRequest::try_from_enc_bandwidth_credential(
+                        enc_credential,
+                        self.shared_key.as_ref().unwrap(),
+                    ) {
+                        Ok(c) => c,
```

Comment:
You can just use `map_err`

Reply:
`map_err` would work if the `handle_initial_authentication_request` function would return a `Result` type. Right now is just a plain enum, I don't really know if it's worth it to refactor that right now